### DISCUSSION TITLE
is vercel using his own database tools ?

### DISCUSSION THREAD

**Discussion by m0x61h0x64i** [vercel/community#1017] [Help] [answered]
i am getting to much database connection errors and these errors are ONLY WITH VERCEL. i tried changing database provider but no fix.

i even get more errors on pro plan!

i want to know is vercel using any related database application or code etc on his own, let me know!

thanks
regards

**Comment by amyegan** [comment, marked as answer]
The most common cause of database connection errors is missing or expired [Environment Variables](https://vercel.com/docs/concepts/projects/environment-variables#). Async code connections can also cause timing errors if not handled carefully. It looks like you already asked a related question in https://github.com/vercel/community/discussions/1016. Let's continue working together to solve your problem in that discussion to keep the whole conversation in one place!
